### PR TITLE
Implement full operator support

### PR DESCRIPTION
**Pull request by lukeeey**
### Introduction
This pull request allows you to change your game mode, the world difficulty, default game mode and game rules from in-game. Additionally, coordinates will now be hidden if the `reducedDebugInfo` game rule (which only exists in java edition) is enabled.

Due to an issue with MCProtocolLib, difficulty changing is currently disabled.

I have shoved a few things in `WorldManager` so i can override it if Geyser is run as a bukkit plugin which probably wasnt the best way to go about it, but im making this PR now for feedback.

### CODE REVIEW

File: bootstrap/bukkit/src/main/java/org/geysermc/platform/bukkit/world/GeyserBukkitWorldManager.java [hunk 1]
```diff
@@ -42,4 +44,14 @@ public BlockState getBlockAt(GeyserSession session, int x, int y, int z) {
         }
         return BlockTranslator.getJavaIdBlockMap().get(Bukkit.getPlayer(session.getPlayerEntity().getUsername()).getWorld().getBlockAt(x, y, z).getBlockData().getAsString());
     }
+
+    @Override
+    public void setGameRule(GeyserSession session, String name, Object value) {
+        Bukkit.getPlayer(session.getPlayerEntity().getUsername()).getWorld().setGameRuleValue(name, (String) value);
```
Comment (lukeeey) [inline bootstrap/bukkit/src/main/java/org/geysermc/platform/bukkit/world/GeyserBukkitWorldManager.java:50]: I actually have no idea about this line and i havent tested it on bukkit yet.

Comment (DoctorMacc) [reply bootstrap/bukkit/src/main/java/org/geysermc/platform/bukkit/world/GeyserBukkitWorldManager.java:49]: This works once PR 490 is applied.

**Comment by lukeeey** [comment]
I’ll make an issue at MCProtocolLib if I remember about the ClientSetDifficultyPacket, I’m out now though so can’t really

**Comment by lukeeey** [comment]
This can be further improved when plugin messaging support is added for Floodgate, as then we dont need to use the hacky method of sending the `/gamemode` or `/defaultgamemode` or `/gamerule` command in chat

**Comment by mtsr** [comment]
I'd be interested in (helping) fixing the `showCoordinates`. What's holding up this PR besides a good rebase? Is there any significant reword needed? Want me to pick it up and finish it?

**Comment by lukeeey** [comment]
It’s not fully done and I’m too busy to work on it

**Comment by Heath123** [comment]
> It’s not fully done and I’m too busy to work on it

They're asking what they need to finish to get it ready to be merged, not why it isn't ready I think

**Comment by mtsr** [comment]
Exactly. I just want to know if and what I can do to help. I'll find my way around the code, mostly need some help with the conceptual side and architecture/design of geyser.

* Is this PR still in line with the design for Geyser or does it need conceptual changes? In particular is the WorldManager approach OK, or can we come up with a different approach?
* Does it need any implementation changes or does it just need finishing?
* Is there other functionality that needs to be added, i.e. more operator functionality?
* Are you OK with me picking up some of the work?

**Comment by RednedEpic** [comment]
Superseded by #1035.

**Pull request closed without merging**